java/java.lsp.server/nbcode/integration/src/org/netbeans/modules/nbcode/integration/LspAssetsDecorationProvider.java

```diff
@@ -101,6 +103,11 @@ public TreeItemData createDecorations(Node n, boolean expanded) {
                 d.addContextValues(CTXVALUE_CAP_REFERENCE_NAME);
                 set = true;
             }
+            
+            if (item instanceof SecretItem) {
```

Comment:
@jhorvath these interface-dependent extractions ("if X in Lookup, use put "prefix:${X.property}" into contextValue) seem rather common / useful ... if the usage will grow, we could invent a declarative support for that, would be then far easier to export state to LSP clients.